app/components/EditProfileForm.tsx

```diff
@@ -1,17 +1,58 @@
 'use client'
 import { FormEvent } from 'react';
+import { useRouter } from 'next/navigation';
 
-export default function EditProfileForm() {
+export default function EditProfileForm({userData}) {
 
-  /**
-   * Submits new post data
-   * 
-   * @param e - the form submit event
-  */
-  const onSubmitHandler = (e: FormEvent<HTMLFormElement>): void => {
+  const router = useRouter();
+
+  const firstName = userData['data'][0]['firstName'][0]
+  const lastName = userData['data'][0]['lastName'][0]
+  const username = userData['data'][0]['username'][0]
+  const gender = userData['data'][0]['gender'][0]
+  const bio = userData['data'][0]['bio'][0]
+
+  const userInfo = new Map([
+    ['firstName', firstName],
+    ['lastName', lastName],
+    ['username', username],
+    ['gender', gender],
+    ['bio', bio]
+  ])
+
+  const onSubmitHandler = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     const formData: Object = Object.fromEntries((new FormData(e.currentTarget)));
-    console.log(JSON.stringify(formData));
+    console.log(formData);
+
+    for (const [key, value] of Object.entries(formData)) {
+      // No support for profile picture yet
+      if (key === "profilePicture") {
+        continue;
+      }
+      // If the value is the same, don't send the request, else send request
+      if (userInfo.get(key) === value) {
+        continue;
+      } else {
+        const payload = {
+          "method" : "PUT",
+          "userInfo" : {
+            "attribute" : key,
+            "newValue" : value
+          }
+        }
+        const res = await fetch(`http://localhost:3000/api/users/${username}`, {
+          method: 'PUT',
+          headers: {'Content-Type': 'application/json'},
+          body: JSON.stringify(payload)
+        });
+
+        const data = await res.json();
+        console.log(data)
+      }
```

Comment:
This works, but I think the logic _might_ be better in the Lambda function?

For example, you just pass in all of the fields and values regardless whether they have been changed or not. Lambda function decides if a specific field has been changed, then it updates the field? This way you don't have to send an individual request for every field that has been changed?

Thoughts?

Reply:
@datderek You're correct, I believe the best way to do it would be to have the endpoint take any number of modifications. The new code should:

1. Take only the modified data from the form and create a JSON request.
2. Send the request to the lambda with all of the modified attributes.
3. Lambda will modify each attribute from the JSON object.

We can still do the change checking client side, I'm just thinking about the API and think that it would be better to make the API support a variable number of attributes instead of just 1 at a time or all of them at once.